py-polars/polars/dataframe/frame.py

```diff
@@ -8283,6 +8298,162 @@ def rows(
         else:
             return self._df.row_tuples()
 
+    def rows_by_key(
+        self,
+        key: str | Sequence[str] | SelectorType,
+        *,
+        named: bool = False,
+        include_key: bool = False,
+        unique: bool = False,
+    ) -> dict[Any, Iterable[Any]]:
+        """
+        Returns DataFrame data as a keyed dictionary of python-native values.
+
+        Parameters
+        ----------
+        key
+            The column(s) to use as the key for the returned dictionary. If multiple
+            columns are specified, the key will be a tuple of those values.
+        named
+            Return dictionary rows instead of tuples, mapping column name to row value.
+        include_key
+            Include key values inline with the associated data (by default the key
+            values are omitted as a memory/performance optimisation, as they can be
+            reoconstructed from the key).
+        unique
+            Indicate that the key is unique; this will result in a 1:1 mapping from
+            key to a single associated row. Note that if the key is *not* actually
+            unique the last row with the given key will be returned.
+
+        Notes
+        -----
+        If you have ``ns``-precision temporal values you should be aware that python
+        natively only supports up to ``us``-precision; if this matters you should export
+        to a different format, as this method returns only python-native values.
+
```

Comment:
Can we add a `See Also: DataFrame::rows`

Reply:
There's already a "See Also" section that includes `rows` - you want an additional reference in the notes?

Reply:
Sorry, missed that. It wondered why we favored this name instead of something like `partition_by` and then I realized this is similar to our `rows`, that's why I want a `See Also`.

If it is already there, it is ok!

Reply:
> this name instead of something like `partition_by` and then I realized this is similar to our `rows`

Yup; I tried a few things, both naming and API-wise. Adding extra params to `rows` (not good!) and a few variations on the name - in the end it seemed to fit best as a standalone addition to the related `*rows*` method(s).